core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -239,37 +251,42 @@ void validateJoinKeySchemas(Map<String, Schema> inputSchemas, Map<String, List<S
 
       Schema schema = inputSchemas.get(stageName);
       if (schema == null) {
-        throw new IllegalArgumentException(String.format("Input schema for input stage %s can not be null", stageName));
+        collector.addFailure(String.format("Input schema for input stage %s cannot be null", stageName), null);
+        collector.getOrThrowException();
       }
 
       for (String joinKey : entry.getValue()) {
         Schema.Field field = schema.getField(joinKey);
         if (field == null) {
-          throw new IllegalArgumentException(String.format("Join key field %s is not present in input of stage %s",
-                                                           joinKey, stageName));
+          collector.addFailure(
+              String.format("Join key field %s is not present in input stage of %s", joinKey, stageName), null)
+              .withConfigProperty(JOIN_KEYS);
+          collector.getOrThrowException();
```

Comment:
don't through here, as we want to collect as many failures as possible.

Reply:
Fixed